packages/react-components/react-nav-preview/library/src/components/AppItem/AppItem.types.ts

```diff
@@ -1,17 +1,37 @@
+import { ARIAButtonSlotProps } from '@fluentui/react-aria';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { NavSize } from '../Nav/Nav.types';
 
 export type AppItemSlots = {
-  root: Slot<'div'>;
+  /**
+   * The root element of the AppItem.
+   */
+  root: NonNullable<Slot<ARIAButtonSlotProps<'a'>>>;
+
+  /**
+   * Icon that renders before the content.
+   */
+  icon?: Slot<'span'>;
 };
 
 /**
  * AppItem Props
  */
-export type AppItemProps = ComponentProps<AppItemSlots> & {};
+export type AppItemProps = ComponentProps<AppItemSlots> & { href?: string };
```

Comment:
I haven't seen the href be added this way via props for button/anchor combo slot in v9 yet - but I really like how it simplifies the as prop usage and type interfaces.

Our slot props function would simply discard this if they force 'as' button had an href applied so it's safe for users too

Reply:
Rock and roll. To be clear, no changes requested on this topic?

Reply:
No changes, maybe a pattern we could use elsewhere for common button/anchor combos